src/AppBundle/Controller/HomepageController.php

```diff
@@ -5,25 +5,36 @@
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\HttpFoundation\Request;
 
 /**
  * @author Jan Klat <jenik@klatys.cz>
  */
 class HomepageController extends Controller
 {
 	/**
-	 * @Route("/", name="homepage")
+	 * @Route("/{page}", name="homepage")
 	 * @Template("homepage/homepage.html.twig")
+     *
+     * @param Request $request
+     * @return array
 	 */
-	public function homepageAction()
+	public function homepageAction(Request $request)
 	{
+        $perPage = 6;
+        $currentPage = $request->attributes->get("page");
+        $totalProducts = $this->getDoctrine()->getRepository(Product::class)->findBy([]);
```

Comment:
Tady pozor, taháš celé produkty jen na to abys je spočítal... teď jich je 44, když jich bude x tisíc, bude to drahé...